common/utils.go

```diff
@@ -115,29 +113,60 @@ func hashWithSalt(password string, salt []byte, iterations int, algorithm string
 	buffer.WriteString(fmt.Sprintf("%s$", algorithm))
 	buffer.WriteString(strconv.Itoa(iterations))
 	buffer.WriteString("$")
-	buffer.WriteString(base64.StdEncoding.EncodeToString(salt))
+	// Re-encode salt, using encoding supplied in saltEncoding param
+	if saltEncoding == "utf-8" {
+		buffer.WriteString(string(salt))
+
+	} else {
+		buffer.WriteString(base64.StdEncoding.EncodeToString(salt))
+	}
 	buffer.WriteString("$")
 	buffer.WriteString(base64.StdEncoding.EncodeToString(hash))
-
+	//log.Debugf("Generated: ", buffer.String())
 	return buffer.String()
 }
 
 // HashCompare verifies that passed password hashes to the same value as the
 // passed passwordHash.
 // Taken from brocaar's lora-app-server: https://github.com/brocaar/lora-app-server
-func HashCompare(password string, passwordHash string) bool {
-	// SPlit the hash string into its parts.
+func HashCompare(password string, passwordHash string, saltEncoding string) bool {
+	//log.Debugf("Supplied:  ", passwordHash)
+	// Split the hash string into its parts.
 	hashSplit := strings.Split(passwordHash, "$")
+	// Check array is of expected length
 	if len(hashSplit) != 5 {
-		log.Error("invalid PBKDF2 hash supplied")
+		log.Errorf("HashCompare, invalid PBKDF2 hash supplied.")
 		return false
 	}
-
-	// Get the iterations and the salt and use them to encode the password
-	// being compared.cre
-	iterations, _ := strconv.Atoi(hashSplit[2])
-	salt, _ := base64.StdEncoding.DecodeString(hashSplit[3])
+	// Define empty error
+	var err error
```

Comment:
No need to.

Reply:
Agreed, this was a legacy of an earlier approach where err was required elsewhere. 
Removed in latest commit.